libraries/chain/include/eosio/chain/wasm_eosio_injection.hpp

```diff
@@ -752,12 +749,19 @@ namespace eosio { namespace chain { namespace wasm_injections {
       using f64_convert_u_i64 = wasm_ops::f64_convert_u_i64       <i64_convert_f64_injector<wasm_ops::f64_convert_u_i64_code>>;
    }; // pre_op_injectors
 
+   struct pre_op_full_injectors : pre_op_injectors {
+      using call_t            = wasm_ops::call                    <call_depth_check_and_insert_checktime>;
+      using call_indirect_t   = wasm_ops::call_indirect           <call_depth_check_and_insert_checktime>;
+   };
 
    struct post_op_injectors : wasm_ops::op_types<pass_injector> {
-      using loop_t   = wasm_ops::loop        <checktime_injection>;
       using call_t   = wasm_ops::call        <fix_call_index>;
    };
 
+   struct post_op_full_injectors : post_op_injectors {
+      using loop_t   = wasm_ops::loop        <checktime_injection>;
+   };
+
```

Comment:
Is there a reason why loop checktime injection is in the post_op_injectors instead of the pre-op injectors?  It would be simpler if you only needed to switch one set of injectors.

Reply:
I'm struggling to remember why; maybe @larryk85 knows.